src/mpyl/steps/run.py

```diff
@@ -82,17 +81,13 @@ def run_properties(self) -> RunProperties:
     def run_plan(self) -> RunPlan:
         return self._run_plan
 
-    @property
-    def has_run_plan_projects(self) -> bool:
-        """
-        We create a ProjectExecution for every project that is changed in the branch we're building, HOWEVER we also
-        know per-project whether it's cached or not. In this function we should read that value to exclude projects
-        that don't need rebuilding.
-        """
-        return not all(
-            len(project_execution) == 0
-            for stage, project_execution in self.run_plan.items()
-        )
+    def has_run_plan_projects(self, include_cached_projects: bool = True) -> bool:
+        for project_executions in self.run_plan.selected_plan.values():
+            for project_execution in project_executions:
+                if include_cached_projects or not project_execution.cached:
+                    return True
+
+        return False
```

Comment:
I think you can do something like
```suggestion
    def has_run_plan_projects(self, include_cached_projects: bool = True) -> bool:
        return any(
            include_cached_projects or not project_execution.cached
            for project_executions in self.run_plan.selected_plan.values()
            for project_execution in project_executions
        )
```